macros/multiple_databases/to_timestamp.sql

```diff
@@ -1,7 +1,12 @@
 {%- macro to_timestamp(field, relation) -%}
 
 {%- if target.type == 'databricks' -%}
-    try_to_timestamp({{ field }}, '{{ var("datetime_format", "yyyy-MM-dd HH:mm:ss[.SSS]") }}')
+    case
+        when try_to_timestamp({{ field }}, '{{ var("datetime_format", "yyyy-MM-dd HH:mm:ss[.SSS]") }}') is not null
+            then try_to_timestamp({{ field }}, '{{ var("datetime_format", "yyyy-MM-dd HH:mm:ss[.SSS]") }}')
+        else
+            try_to_timestamp({{ field }}, 'HH:mm:ss[.SSS]')
+    end
 {# Snowflake try_to function requires an expression of type varchar. #}
```

Comment:
I think changing this should be out of scope for this PR. This requires at least more testing/discussion, which is the reason we also made it a separate task. The other functions can be changed without considering this one already.

Reply:
Agree, I'll scope it out